src/Microsoft.Framework.Configuration.Binder/ConfigurationBinder.cs

```diff
@@ -68,9 +68,9 @@ private static void BindProperty(PropertyInfo property, object propertyOwner, IC
             }
         }
 
-        private static object BindType(Type type, object typeInstance, IConfiguration configuration)
+        private static object BindType(Type type, object typeInstance, IConfigurationSection configuration)
         {
-            var configValue = configuration.Get(null);
+            var configValue = configuration.Value;
```

Comment:
So when I originally wrote this forever ago, my thinking was that the callee takes care of figuring out what part of the config tree maps to the object, and this code just binds against what its given.  

Seems like it should be valid to bind against the entire config tree, as well as drilling down to a specific section and then binding.